geopandas/tools/sjoin.py

```diff
@@ -329,3 +330,178 @@ def _frame_join(indices, left_df, right_df, how, lsuffix, rsuffix):
+        indices = pd.DataFrame(
+            {"_key_left": l_idx, "_key_right": r_idx, "distances": distances}
+        )
+    else:
+        # when sindex is empty / has no valid geometries
+        indices = pd.DataFrame(
+            columns=["_key_left", "_key_right", "distances"], dtype=float
+        )
+    return indices
+
+
+def sjoin_nearest(
+    left_df,
+    right_df,
+    how="inner",
+    max_distance=None,
+    lsuffix="left",
+    rsuffix="right",
+    distance_col=None,
+) -> GeoDataFrame:
+    """Spatial join of two GeoDataFrames based on the distance between their geometries.
+
+    See the User Guide page :doc:`../../user_guide/mergingdata` for details.
+
+
+    Parameters
+    ----------
+    left_df, right_df : GeoDataFrames
+    how : string, default 'inner'
+        The type of join:
+
+        * 'left': use keys from left_df; retain only left_df geometry column
+        * 'right': use keys from right_df; retain only right_df geometry column
+        * 'inner': use intersection of keys from both dfs; retain only
+          left_df geometry column
+    max_distance : float, default None
+        Maximum distance within which to query for nearest geometry.
+        Must be greater than 0.
+        The max_distance used to search for nearest items in the tree may have a
+        significant impact on performance by reducing the number of input
+        geometries that are evaluated for nearest items in the tree.
+    lsuffix : string, default 'left'
+        Suffix to apply to overlapping column names (left GeoDataFrame).
+    rsuffix : string, default 'right'
+        Suffix to apply to overlapping column names (right GeoDataFrame).
+    distance_col : string, default None
+        If set, save the distances computed between matching geometries under a
+        column of this name in the joined GeoDataFrame.
+
+    Examples
+    --------
+    >>> countries = geopandas.read_file(geopandas.datasets.get_\
+path("naturalearth_lowres"))
+    >>> cities = geopandas.read_file(geopandas.datasets.get_path("naturalearth_cities"))
+    >>> countries.head()  # doctest: +SKIP
+        pop_est      continent                      name \
+iso_a3  gdp_md_est                                           geometry
+    0     920938        Oceania                      Fiji    FJI      8374.0  MULTIPOLY\
+GON (((180.00000 -16.06713, 180.00000...
+    1   53950935         Africa                  Tanzania    TZA    150600.0  POLYGON (\
+(33.90371 -0.95000, 34.07262 -1.05982...
+    2     603253         Africa                 W. Sahara    ESH       906.5  POLYGON (\
+(-8.66559 27.65643, -8.66512 27.58948...
+    3   35623680  North America                    Canada    CAN   1674000.0  MULTIPOLY\
+GON (((-122.84000 49.00000, -122.9742...
+    4  326625791  North America  United States of America    USA  18560000.0  MULTIPOLY\
+GON (((-122.84000 49.00000, -120.0000...
+    >>> cities.head()  # doctest: +SKIP
```

Comment:
```suggestion
    >>> cities.head(2)  # doctest: +SKIP
```

Again, this is quite noisy with 5 rows.